lib/path.js

```diff
@@ -616,10 +628,11 @@ const win32 = {
   },
 
   /**
-   * @param {string} path
+   * @param {PathLike} path
    * @returns {string}
    */
   toNamespacedPath(path) {
+    path = toPathIfFileURL(path, { windows: true });
```

Comment:
Creating an object is almost costly. We can use bitwise operations to define attributes instead of constructing a boolean and an object for this scenario. A similar approach was used recently with streams.